execute_test.go

```diff
@@ -2095,3 +2095,181 @@ func TestSingleObjectWithColonInID(t *testing.T) {
 		"id": "Thing:1337", "firstName": "Foo", "lastName": "bar",
 	}, value)
 }
+
+// TestExecutor_plansWithManyDeepDependencies test that two `Then` works without races
+// Test provides quite deep query to illustrate problem with
+// reused memory during slice `appends`
+func TestExecutor_plansWithManyDeepDependencies(t *testing.T) {
+
+	// the query we want to execute is
+	//	{
+	//		user {						<- serviceA
+	//			parent {				<- serviceA
+	//				parent {			<- serviceA
+	//					id  			<- serviceA
+	//					house {			<- serviceB
+	//						address		<- serviceC
+	//						cats {		<- serviceC
+	//							id  	<- serviceC
+	//							name	<- serviceD
+	//						}
+	//					}
+	//				}
+	//			}
+	//		}
+	//	}
+
+	// test helpers
+	definitionFactory := func(s string) *ast.FieldDefinition {
```

Comment:
@AlecAivazis writing test plan is quite repetitive and easy to make mistake. Maybe some factories & test utils will be helpful to optimize that process.

Reply:
yea, you're very correct haha. i have been telling myself to clean that up but have always gotten distracted by other things. If you have any ideas for utilities, im definitely interested to hear more :)